### ISSUE TITLE
Space between functions (class methods)

### ISSUE THREAD

**Issue by ScottWaring** [eslint/eslint#14582]
**Tell us about your environment**

* **ESLint Version:** 7.25.0
* **Node Version:** 10.15.1
* **npm Version:** 6.8.0

**Parsers**
@babel/eslint-parser": "^7.13.14",
"@typescript-eslint/eslint-plugin": "^4.22.1",
"@typescript-eslint/parser": "^4.22.1",

AngularJS: 1.6.3
Our `.eslintrc.json` file
```
{
    "env": {
        "browser": true,
        "node": true,
        "commonjs": true,
        "es6": true,
        "jasmine": true
    },
    "extends": ["google"],
    "rules": {
        "indent": ["error", 4, { "SwitchCase": 1 }],
        "no-var": "off",
        "prefer-const": "off",
        "no-unused-vars": "off",
        "require-jsdoc": "off",
        "max-len": "off",
        "new-cap": ["error", {"capIsNewExceptions": ["DataTable"]}],
        "valid-jsdoc": "off",
        "comma-dangle": "off",
        "no-debugger": "error",
        "no-extra-semi": "error",
        "no-unneeded-ternary": "error",
        "prefer-object-spread": "warn",
        "no-multiple-empty-lines": ["error", {"max": 1}],
        "space-infix-ops": "error",
        "lines-around-comment": ["error", { "beforeBlockComment": true, "allowBlockStart": true }],
        "space-before-blocks": ["error", {"functions": "always"}],
        "lines-between-class-members": ["error", "always", { "exceptAfterSingleLine": true }],
        "padding-line-between-statements": [
            "error",
            { "blankLine": "always", "prev": "directive", "next": "*" },
            { "blankLine": "always", "prev": "*", "next": ["class", "function", "export"] },
            { "blankLine": "any", "prev": "directive", "next": "directive" },
            { "blankLine": "any", "prev": "export", "next": "export" },
            { "blankLine": "always", "prev": "block", "next": "block" }
        ]
    },
    "globals": {
        "$": false,
        "moment": false,
        "Twilio": false,
        "angular":false,
        "base64":false,
        "swal": false,
        "firebase": false,
        "zE": false,
        "NProgress": false,
        "TweenMax": false,
        "Linear": false,
        "cleanUI": false,
        "Ladda": false,
        "Typer": false,
        "Chart": false
    },
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "sourceType": "module"
    },
    "plugins": ["@typescript-eslint"]
}
```

I want eslint to error when there is no space between two function blocks. 

```js
export class DoSomething {    
    private functionFoo() {
         // ....
    } // I want eslint to error when encountering this
    private functionBar() {
        // ......  
    }
}
```

</details>

These rules do not error when encountering those functions.
```
"space-before-blocks": ["error", {"functions": "always"}],
 "padding-line-between-statements": [
            "error",
            { "blankLine": "always", "prev": "directive", "next": "*" },
            { "blankLine": "always", "prev": "*", "next": ["class", "function", "export"] },
            { "blankLine": "any", "prev": "directive", "next": "directive" },
            { "blankLine": "any", "prev": "export", "next": "export" },
            { "blankLine": "always", "prev": "block", "next": "block" }
        ]
```

The rule that comes closest to what I want, creates a different issue.
```
lines-between-class-members: ["error", "always", { "exceptAfterSingleLine": true }]
```

```js
export class DoSomething {    
   private varFoo: string;
   private varBar: {
        key: something
    } // It errors here which is undesired
   private varSomething: boolean;

   constructor() {}

    private functionFoo() {
         // ....
    } // It errors here as desired
    private functionBar() {
        // ......  
    }
}
```

If anyone has insight in to how I can configure our eslint to behave as we want - it is greatly appreciated!

**Comment by ScottWaring** [comment]
We have the our typescript parser, and the rules 
```
      "lines-between-class-members": "off",
        "@typescript-eslint/lines-between-class-members": ["error", "always", { "exceptAfterOverload": true, "exceptAfterSingleLine": true }]
````
behave the exact same way as the eslint rules do, which is to say not our desired behavior

**Comment by ljharb** [comment]
What I'm saying is that for the issue to be relevant to this repo, it would need to be using the default parser and valid JS code for it - at which point it'd be a stylistic change that's unlikely to get accepted.

For the TS change, you'd need to file an issue on the typescript-eslint repo.

**Comment by nzakas** [comment]
As @ljharb mentioned, this is TypeScript-related, so visiting typescript-eslint is the best path forward.
